apps/challenges/aws_utils.py

```diff
@@ -0,0 +1,301 @@
+                    "value": "{queue_name}"
+                }},
+            ],
+            "workingDirectory": "/code",
+            "readonlyRootFilesystem": False,
+            "logConfiguration": {{
+                "logDriver": "awslogs",
+                "options": {{
+                    "awslogs-create-group": "true",
+                    "awslogs-region": "us-east-1",
+                    "awslogs-group": "{log_group}",
+                    "awslogs-stream-prefix": "{queue_name}"  # log stream format: prefix-name/container-name/ecs-task-id
+                }},
+            }},
+            "healthCheck": {{
+                "command": [  # Need healthcheck command.
+                    ""
+                ],
+                "interval": 30,
+                "timeout": 5,
+                "retries": 3,
+                "startPeriod": 0
+            }},
+        }}
+    ],
+    "requiresCompatibilities":[
+        "FARGATE"
+    ],
+    "cpu":"",  # Need hard limits.
+    "memory":"",
+}}
+"""
+
+service_definition = """
+{{
+    "cluster":"Challenge_Cluster",
+    "serviceName":"{service_name}",
+    "taskDefinition":"{task_def_arn}",
+    "desiredCount":1,
+    "clientToken":"{client_token}",
+    "launchType":"FARGATE",
+    "platformVersion":"LATEST",
+    "networkConfiguration":{{ # Need to create VPC before filling this.
+        "awsvpcConfiguration": {{
+            "subnets": [
+                # "string",
+            ],
+            'securityGroups': [
+                "string",
+            ],
+            "assignPublicIp": "ENABLED"
+        }}
+    }},
+    "healthCheckGracePeriodSeconds":30,
+    "schedulingStrategy":"REPLICA",
+    "deploymentController":{{
+        "type": "ECS"
+    }},
+    "enableECSManagedTags":True,
+    "propagateTags":"SERVICE"
+}}
+"""
+
+update_service_args = """
+{{
+    "cluster":"Challenge_Cluster",
+    "service":"{service_name}",
+    "desiredCount":num_of_tasks,
+    "taskDefinition":"{task_def_arn}",
+    "forceNewDeployment":False
+}}
+"""
+
+aws_keys = {
+    "AWS_ACCOUNT_ID": os.environ.get("AWS_ACCOUNT_ID", "x"),
+    "AWS_ACCESS_KEY_ID": os.environ.get("AWS_ACCESS_KEY_ID", "x"),
+    "AWS_SECRET_ACCESS_KEY": os.environ.get("AWS_SECRET_ACCESS_KEY", "x"),
+    "AWS_REGION": os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+}
+
+
+def client_token_generator():
+    client_token = ''.join(random.choices(string.ascii_letters + string.digits, k=32))
+    return client_token
+
+"""
+Called by: create_service_by_challenge_pk (below).
+
+Registers the task definition for a new challenge, and raises a custom exception
+if the worker service is already active (which is caught by the create_service custom method.).
+"""
+
+
+def register_task_def_by_challenge_pk(client, queue_name, challenge):
+    container_name = "worker_{}".format(queue_name)
+    image = "{AWS_ACCOUNT_ID}.dkr.ecr.us-east-1.amazonaws.com/evalai-production-worker:latest".format(AWS_ACCOUNT_ID=aws_keys["AWS_ACCOUNT_ID"])
```

Comment:
This code will fail if we use it in the dev mode or in staging since you have hard coded the repository name i.e `evalai-production-worker`. Instead, you should pull from the environment.

Reply:
Done.